core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -193,6 +194,13 @@ public RemoteClusterService getRemoteClusterService() {
         return transportService.getRemoteClusterService();
     }
 
+    /**
+     * Return a map of nodeId to pending number of requests for the given action name prefix
+     */
+    public Map<String, Long> getPendingRequests(final String actionNamePrefix) {
```

Comment:
> ideally pre-filled with every possible action name so that the map is effectively immutable afterwards and concurrency is only handled at the AtomicLong level? It would also create fewer boxed longs.

This is a map of `nodeId` to `connectionCount`, I'm not sure how pre-filling it with possible action names would help?